### PR TITLE
Backport fixes from master and clean up before new release

### PR DESCRIPTION
**Pull request by kgaillot**
Includes fix for CLBZ#5384

### CODE REVIEW

File: lib/common/utils.c [hunk 1]
```diff
@@ -743,7 +743,7 @@ crm_pid_active(long pid, const char *daemon)
     } else if (rc == 0 && (daemon == NULL || have_proc_pid == -1)) {
         return 1;  /* kill as the only indicator, cannot double check */
 
-    } else {
+    } else if (daemon != NULL) {
```
Comment (jnpkrn) [inline lib/common/utils.c]: Good spot but semantically incorrect.
needs to be:
```diff
- } else if (rc < 0 && have_proc_pid == -1) {
+ } else if (rc < 0 && (have_proc_pid == -1 || daemon == NULL)) {
```
Please, fix master as well, or tell me.

Comment (jnpkrn) [reply lib/common/utils.c]: Also note that the only scenario applicable for the very unlikely
progress resulting in segfault is via `crm_procfs_pid_of` that is
just as well conceptially broken as with the recent CVE, but I wanted
to keep the scope at least partially limited, as the code makes
bad assumptions throughout...

Comment (kgaillot) [reply lib/common/utils.c]: Thanks, if c18c3f1 from #1764 looks good, I'll backport it here

Comment (jnpkrn) [reply lib/common/utils.c]: Yes, please.  Thanks!

Comment (kgaillot) [reply lib/common/utils.c]: Updated

**Pull request merged**